vos/vos/commonparser.py

```diff
@@ -1,92 +1,89 @@
-        self.add_option("-v", "--verbose", action="store_true", default=False,
-                        help="Print verbose level log messages")
-        self.add_option("-w", "--warning", action="store_true", default=False,
-                        help="Print warning level log messages")
-
-        self.epilog =\
-"""The vos commands use a configuration file for user specific settings.
-A user can edit this file to change the behaviour of the vos commands.
-The configuration is typically found in the user's home directory in .config/vos/vos-config.
-
-The file contains sections with settings pertaining to that section. The current setting are:
-
-[vos]
-# The default vospace authority
-resourceID = ivo://cadc.nrc.ca/vospace
-
-[transfer]
-# The default transfer protocol
-protocol = http
-"""
-
-    def format_epilog(self, formatter):
-        return self.epilog
 
-    def process_informational_options(self):
-        """Display version, set logging verbosity"""
-        (opt, args) = self.parse_args()
+def set_logging_level_from_args(args):
+    """Display version, set logging verbosity"""
 
-        if opt.version:
-            self.print_version()
-            sys.exit(0)
+    # Logger verbosity
+    if args.debug:
+        args.log_level = logging.DEBUG
+    elif args.verbose:
+        args.log_level = logging.INFO
+    elif args.warning:
+        args.log_level = logging.WARNING
+    else:
+        args.log_level = logging.ERROR
 
-        # Logger verbosity
-        if opt.debug:
-            self.log_level = logging.DEBUG
-        elif opt.verbose:
-            self.log_level = logging.INFO
-        elif opt.warning:
-            self.log_level = logging.WARNING
-        else:
-            self.log_level = logging.ERROR
+    log_format = "%(levelname)s %(module)s %(message)s"
+    if args.log_level < logging.INFO:
+        log_format = ("%(levelname)s %(asctime)s %(thread)d vos-" + str(version) +
+                      " %(module)s.%(funcName)s.%(lineno)d %(message)s")
+    logging.basicConfig(format=log_format, level=args.log_level)
+    logger = logging.getLogger('root')
+    logger.setLevel(args.log_level)
 
-        log_format = "%(levelname)s %(module)s %(message)s"
-        if self.log_level < logging.INFO:
-            log_format = ("%(levelname)s %(asctime)s %(thread)d vos-"+str(version)+" %(module)s.%(funcName)s.%(lineno)d %(message)s")
-        logging.basicConfig(format=log_format, level=self.log_level)
+    if args.vos_debug:
+        logger = logging.getLogger('vos')
+        logger.setLevel(logging.DEBUG)
 
-        if opt.vos_debug:
-            logger = logging.getLogger('vos')
-            logger.setLevel(logging.DEBUG)
+    if sys.version_info[1] > 6:
+        logging.getLogger().addHandler(logging.NullHandler())
 
-        if sys.version_info[1] > 6:
-            logger = logging.getLogger()
-            logger.addHandler(logging.NullHandler())
 
+class CommonParser(argparse.ArgumentParser):
```

Comment:
Too bad we could not come to a common approach with the rest of the CADC apps. In essence, this does the same thing (although differently) with the code in cadcutils.